edx_proctoring/models.py

```diff
@@ -321,6 +324,58 @@ def get_active_student_attempts(self, user_id, course_id=None):
 
         return self.filter(filtered_query).order_by('-created')  # pylint: disable=no-member
 
+    def get_last_relevant_proctored_practice_attempt(self, user_id, proctoring_backend, course_id=None):
+        """
+        Filter user's proctored practice attempts within the last two years, and return one of the
+        following:
+            1. The last proctored practice attempt, if it exists
+            2. The most recent attempt, if there is no verified attempt
+            3. None, if there are no proctored practice attempts
+        
+        Parameters:
+            * user_id: ID of the user
+            * proctoring_backend: The name of the proctoring backend that is being used
+            * course_id: Optional, used if we want to prioritize attempts from a certain course
+                         (I.E. for the instructor dashboard)
+        """
+        earliest_allowed_date = datetime.now(pytz.UTC) - timedelta(days=730)
+        proctored_practice_attempts = self.filter(
+            user_id=user_id, taking_as_proctored=True, proctored_exam__is_practice_exam=True,
+            proctored_exam__backend=proctoring_backend, modified__gt=earliest_allowed_date
+        ).order_by('-modified')
+
+        if proctored_practice_attempts.exists():
```

Comment:
I don't believe this if statement is needed. This is because, we use the records in `proctored_practice_attempts` below to determine behavior anyways. Filtering on top of this result, if this result is already empty, will return empty anyways.